File: 백준/Silver/10845. 큐/큐.cc
#include <utility>      
#include <iostream>
#include <queue>
#include <string>
int main()
{
    std::ios::sync_with_stdio(0);
    std::cin.tie(0);
    int n{}, num{};
    std::string input;
    std::queue <int> que{};
    std::cin >> n;
    for (int i=0; i < n; i++){
        std::cin >> input;
        if ( input == "push"){
            std::cin >> num;
            que.push(num);
        }
        else if (input =="size"){
            std::cout << que.size() << "\n";
        }
        else if (input == "empty"){
            if(que.empty() == true){
                std::cout << 1 << "\n";
            }
            else std::cout << 0<< "\n";
        }
        else if(que.empty() == true){
            std::cout << -1<< "\n";
        }
        else if(input == "pop"){
            std::cout << que.front()<< "\n";
            que.pop();
        } 
        else if(input == "front") std::cout << que.front()<< "\n";
        else if(input == "back") std::cout << que.back()<< "\n";
    }
}

File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque
n=int(sys.stdin.readline())
arr= deque()
for i in range(n):
    com=sys.stdin.readline().split()
    if(com[0]=="push"):
        arr.append(int(com[1]))
    elif(com[0]=="pop"):
        if(arr):
            print(arr.popleft())
        else:
            print(-1)
    elif(com[0]=="size"):
        print(len(arr))
    elif(com[0]=="empty"):
        if(arr):
            print(0)
        else:
            print(1)
    elif(com[0]=="front"):
        if(arr):
            print(arr[0])
        else:
            print(-1)
    elif(com[0]=="back"):
        if(arr):
            print(arr[-1])
        else:
            print(-1)